### PR TITLE
Add navigation favorites tests

### LINKED ISSUE #6406: Favorites class needs unit tests
The `Favorite` class could use some basic tests:

* `add_item` - Also check for correct error responses
* `remove_item` - Also check for correct error responses
* `get_all`

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #6406 

Adds some unit tests around the Favorites class.

### Detailed test instructions:

1. Check that tests pass ✅ 

`npm run test:php -- --filter=WC_Tests_Navigation_Favorites`

### CODE REVIEW

File: tests/navigation/class-wc-favorites.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		$favorites = $this->instance->get_all();
+		$this->assertContains( 'item-to-remove', $favorites );
+
+		$result    = $this->instance->remove_item( 'item-to-remove' );
+		$favorites = $this->instance->get_all();
+		$this->assertNotContains( 'item-to-remove', $favorites );
+	}
+
+	/**
+	 * Test that favorites can be added.
```
Comment (ilyasfoo) [inline tests/navigation/class-wc-favorites.php:82]: Comment seems off here

**Review by ilyasfoo — APPROVED** [review]
Left a minor comment on function docs. Pre-approved 🚢 !

File: tests/navigation/class-wc-favorites.php (hunk 1 above)
Comment (joshuatf) [reply tests/navigation/class-wc-favorites.php]: Good catch, @ilyasfoo!  Updated.

Comment (ilyasfoo) [reply tests/navigation/class-wc-favorites.php]: LGTM!

**Pull request merged**